src/main/java/org/kuali/kra/proposaldevelopment/web/struts/action/ProposalDevelopmentActionsAction.java

```diff
@@ -308,37 +311,51 @@ private boolean hasAskedToNotReceiveFutureRequests( WorkflowDocument workflowDoc
 				if ( variableKey.startsWith( KewApiConstants.RECEIVE_FUTURE_REQUESTS_BRANCH_STATE_KEY )
 						&& variableValue.toUpperCase().equals( KewApiConstants.RECEIVE_FUTURE_REQUESTS_BRANCH_STATE_VALUE )
 						&& variableKey.contains( principalId ) ) {
+					LOG.info("hasAskedToReceiveFutureRequests: receiveFutureRequests = true");
 					receiveFutureRequests = true;
-					break;
 				}
 				else if ( variableKey.startsWith( KewApiConstants.RECEIVE_FUTURE_REQUESTS_BRANCH_STATE_KEY )
 						&& variableValue.toUpperCase().equals( KewApiConstants.DONT_RECEIVE_FUTURE_REQUESTS_BRANCH_STATE_VALUE )
 						&& variableKey.contains( principalId ) ) {
+					LOG.info("hasAskedToReceiveFutureRequests: doNotReceiveFutureRequests = true");
 					doNotReceiveFutureRequests = true;
 					break;
 				}
 			}
 		}
 
-		return ( receiveFutureRequests == false && doNotReceiveFutureRequests == false );
+		//has to account for both cases, when the user first wants to receive further requests, but then later on decides he doesn't want to
+		LOG.info("hasAskedToReceiveFutureRequests: returning ="+ (!doNotReceiveFutureRequests || receiveFutureRequests));
+		return ( !doNotReceiveFutureRequests || receiveFutureRequests);
 	}
 
 	private boolean canGenerateMultipleApprovalRequests( RoutingReportCriteria reportCriteria, String loggedInPrincipalId, String currentRouteNodeNames ) throws Exception {
-		int approvalRequestsCount = 0;
 		WorkflowDocumentActionsService info = GlobalResourceLoader.getService( "rice.kew.workflowDocumentActionsService" );
-
 		DocumentDetail results1 = info.executeSimulation( reportCriteria );
+		int crtUserActiveActionRequestsCount = 0;
 		for ( ActionRequest actionRequest : results1.getActionRequests() ) {
-			// !actionRequest.isRoleRequest() && removed from condition for
-			if ( actionRequest.isPending() && actionRequest.getActionRequested().getCode().equalsIgnoreCase( KewApiConstants.ACTION_REQUEST_APPROVE_REQ ) &&
-					recipientMatchesUser( actionRequest, loggedInPrincipalId ) && !StringUtils.contains( currentRouteNodeNames, actionRequest.getNodeName() ) ) {
-				approvalRequestsCount += 1;
+			if ( actionRequest.isPending()
+			        && actionRequest.getActionRequested().getCode().equalsIgnoreCase( KewApiConstants.ACTION_REQUEST_APPROVE_REQ ) 
+			        && recipientMatchesUser( actionRequest, loggedInPrincipalId ) )	{
+				crtUserActiveActionRequestsCount++;
+				LOG.info("MultipleApprovalRequests: actionRequestId="+actionRequest.getId()+" actionRequestAnnotation="+actionRequest.getAnnotation()+" crtUserActiveActionRequests="+crtUserActiveActionRequestsCount);
```

Comment:
Should this be LOG.debug instead? Any concerns about the number of log entries caused by the LOG.info?